src/Shared/LayoutRenderers/AspNetRequestItemLayoutRenderer.cs

```diff
@@ -0,0 +1,71 @@
+using System.Text;
+using NLog.LayoutRenderers;
+using NLog.Config;
+#if !ASP_NET_CORE
+using System.Web;
+#else
+using Microsoft.AspNetCore.Http;
+#endif
+
+namespace NLog.Web.LayoutRenderers
+{
+    /// <summary>
+    /// ASP.NET HttpContext Item.
+    /// </summary>
+    /// <remarks>
+    /// Use this layout renderer to insert the value of the specified Item of the HttpContext
+    /// </remarks>
+    /// <example>
+    /// <para>Example usage of ${aspnet-httpcontext-item}:</para>
+    /// <code lang="NLog Layout Renderer">
+    /// ${aspnet-httpcontext-item:Item=v}
+    /// </code>
+    /// </example>
+    [LayoutRenderer("aspnet-httpcontext-item")]
+    public class AspNetRequestItemLayoutRenderer : AspNetLayoutRendererBase
+    {
+        /// <summary>
+        /// Gets or sets the item name. The Item collection variables having the specified name are rendered.
+        /// </summary>
+        [RequiredParameter]
+        [DefaultParameter]
+        public string Item { get; set; }
+
+        /// <summary>
+        /// Renders the specified ASP.NET HttpContext.Item and appends it to the specified <see cref="StringBuilder" />.
```

Comment:
Please use `<inheritdoc/>` and keep relevant details in class-docs.

Reply:
Complete